fusor-ember-cli/app/components/tr-subscription-manifest.js

```diff
@@ -2,4 +2,22 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
   tagName: 'tr',
+
+  qtyColumn: Ember.computed('isDisconnected',
+                            'subscription.quantity_to_add',
+                            'subscription.quantity_attached',
+                            'subscription.qtySumAttached', function() {
+      if (this.get('isDisconnected')) {
+          return this.get('subscription.quantity_attached');
+
+      } else if (this.get('subscription.quantity_to_add') > 0) {
+          return this.get('subscription.quantity_attached') + ' + ' +
+                 this.get('subscription.quantity_to_add') + ' = ' +
+                 this.get('subscription.qtySumAttached');
+      } else {
+          return this.get('subscription.quantity_attached');
+      }
+
```

Comment:
Good point. I agreed that there are places that can be more DRY.
On Dec 17, 2015 5:36 PM, "Christopher Chase" <notifications@github.com>
wrote:

> In fusor-ember-cli/app/components/tr-subscription-manifest.js
> <https://github.com/fusor/fusor/pull/577#discussion_r47919490>:
>
> > +
> > +  qtyColumn: Ember.computed('isDisconnected',
> > +                            'subscription.quantity_to_add',
> > +                            'subscription.quantity_attached',
> > +                            'subscription.qtySumAttached', function() {
> > +      if (this.get('isDisconnected')) {
> > +          return this.get('subscription.quantity_attached');
> > +
> > +      } else if (this.get('subscription.quantity_to_add') > 0) {
> > +          return this.get('subscription.quantity_attached') + ' + ' +
> > +                 this.get('subscription.quantity_to_add') + ' = ' +
> > +                 this.get('subscription.qtySumAttached');
> > +      } else {
> > +          return this.get('subscription.quantity_attached');
> > +      }
> > +
>
> Ok. Looks like 2 places returning the same thing. You could refactor it to
> look something like this if you wanted:
>
>       var toAdd = this.get('subscription.quantity_to_add'),
>         attached = this.get('subscription.quantity_attached'),
>         sum = this.get('subscription.qtySumAttached');
>
>       if (!this.get('isDisconnected') && toAdd > 0) {
>         return `${attached} + ${toAdd} = ${sum}`;
>       }
>
>       return  attached;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fusor/fusor/pull/577/files#r47919490>.
>